src/Composer/Installer.php

```diff
@@ -0,0 +1,209 @@
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\ManagerPlugin;
+
+/**
+ * This class is generated by contao/manager-bundle, specifically by
+ * @see \Contao\ManagerBundle\Composer\Installer
+ *
+ * This file is overwritten at every run of `composer install` or `composer update`.
+ */
+%s
+{
+    const BUNDLE_PLUGINS = 'Contao\ManagerPlugin\Bundle\BundlePluginInterface';
+    const CONFIG_PLUGINS = 'Contao\ManagerPlugin\Config\ConfigPluginInterface';
+    const EXTENSION_PLUGINS = 'Contao\ManagerPlugin\Config\ExtensionPluginInterface';
+    const ROUTING_PLUGINS = 'Contao\ManagerPlugin\Routing\RoutingPluginInterface';
+
+    /**
+     * @var array
+     */
+    private $plugins;
+
+    /**
+     * Constructor.
+     */
+    public function __construct()
+    {
+        $this->plugins = %s;
+    }
+
+    /**
+     * Returns all active plugin instances.
+     *
+     * @return array
+     */
+    public function getInstances()
+    {
+        return $this->plugins;
+    }
+
+    /**
+     * Returns the active plugin instances of a given type (see class constants).
+     *
+     * @param string $type
+     * @param bool   $reverseOrder
+     *
+     * @return array
+     */
+    public function getInstancesOf($type, $reverseOrder = false)
+    {
+        $plugins = array_filter(
+            $this->getInstances(),
+            function ($plugin) use ($type) {
+                return is_a($plugin, $type);
+            }
+        );
+
+        $plugins = $reverseOrder ? array_reverse($plugins, true) : $plugins;
+
+        return $plugins;
+    }
+}
+
+PHP;
+
+    /**
+     * Sets the Contao Manager plugins.
+     *
+     * @param Locker               $locker
+     * @param RootPackageInterface $rootPackage
+     *
+     * @throws \RuntimeException
+     */
+    public function dumpPlugins(Locker $locker, RootPackageInterface $rootPackage)
+    {
+        $plugins = [];
+
+        $disabled = $this->getDisabledPackages($rootPackage);
+        $lockData = $locker->getLockData();
+        $lockData['packages-dev'] = isset($lockData['packages-dev']) ? $lockData['packages-dev'] : [];
+
+        foreach (array_merge($lockData['packages'], $lockData['packages-dev']) as $package) {
+            if (in_array($package['name'], $disabled, true)) {
+                continue;
+            }
+
+            if (isset($package['extra']['contao-manager-plugin'])) {
+                if (!class_exists($package['extra']['contao-manager-plugin'])) {
+                    throw new \RuntimeException(
+                        sprintf('Plugin class "%s" not found', $package['extra']['contao-manager-plugin'])
+                    );
+                }
+
+                $plugins[$package['name']] = new $package['extra']['contao-manager-plugin']();
+            }
+        }
+
+        $plugins = $this->orderPlugins($plugins);
+
+        // Instantiate a global plugin to load AppBundle or other customizations
+        $appPlugin = '\ContaoManagerPlugin';
```

Comment:
Shouldn't this be in the `App\` namespace?

Reply:
no, it is the global namespace. We usually add this class to the `app` folder (see https://github.com/terminal42/contao-standard/tree/master/app)

Reply:
> We usually add this class to the `app` folder

Exactly this is why it should be in the `App\` namespace. 😄

Reply:
But it has been in the global namespace ever since, so never mind.

Reply:
> But it has been in the global namespace ever since, so never mind.

Correct. Also, the `App\` namespace is in `src/` folder and not `app/`. People following the Sf best practice might have an `AppBundle` instead etc.

Reply:
True.